base/strings/util.jl

```diff
@@ -572,6 +590,13 @@ If `pat` is a regular expression and `r` is a [`SubstitutionString`](@ref), then
 references in `r` are replaced with the corresponding matched text.
 To remove instances of `pat` from `string`, set `r` to the empty `String` (`""`).
 
+Multiple patterns can be specified, and they will be applied left-to-right
+simultaneously, so only one pattern will be applied to any character, and the
+patterns will only be applied to the input text, not the replacements.
```

Comment:
I think this section is misleading - the text seems to sugest only patterns of the form `'x' => "replacement"` or `'x' => 'a'` are allowed, but the tests further down seem to suggest `"original" => "replacement"`  and `"original" => 'a'` should also work.

Additionally, what should the result of `replace("atest", "atest" => 'a', "at" => 'b')` be? Simply `"a"`, since the patterns are applied left to right, even though "simultaneously" applying them leads to inconsistency (and imo both are reasonable choices)?

Reply:
The patterns are "not applied to the replacements". Each character is only matched by at most one pattern, though a pattern may match multiple characters (or even none).